### PR TITLE
Version 12.0.0 patch queued-request-controller

### PR DESCRIPTION
**Pull request by jiexi**
Core PR: https://github.com/MetaMask/core/pull/4423
Patch Core Branch: jl/patch-queued-request-controller@0.10.0-replace-methods-array-with-callback
Extension PR: https://github.com/MetaMask/metamask-extension/pull/25310

Fixes issue with `eth_requestAccounts` always being treated like it will prompt a confirmation when that isn't always true

https://github.com/MetaMask/metamask-extension/assets/918701/65558463-0927-44dc-a0d0-172ec45c76ba

### CODE REVIEW

File: .yarn/patches/@metamask-queued-request-controller-npm-0.10.0-a228bc1b3b.patch [hunk 1]
```diff
@@ -0,0 +1,1508 @@
+diff --git a/dist/QueuedRequestController.js b/dist/QueuedRequestController.js
```
Comment (adonesky1) [inline .yarn/patches/@metamask-queued-request-controller-npm-0.10.0-a228bc1b3b.patch:1]: This patch looks like it contains a bunch more than it needs to? Seems to include way more than just the lines that are [changed in the core side PR](https://github.com/MetaMask/core/pull/4423)?

Comment (jiexi) [reply .yarn/patches/@metamask-queued-request-controller-npm-0.10.0-a228bc1b3b.patch:1]: the outputted chunk names are entirely different depending on the contents of the chunk, and the diff is not able to reconcile this as a file rename + line changes, instead calling it a file delete and a file create

Comment (adonesky1) [reply .yarn/patches/@metamask-queued-request-controller-npm-0.10.0-a228bc1b3b.patch:1]: Linking [this thread](https://consensys.slack.com/archives/CTQAGKY5V/p1718123750012709) on why this kindof (full override) patch for posterity

**Comment by adonesky1** [comment]
Verified that this fixes the bug we set out to fix. But did run into a very strange behavior while messing around that I will try now to reproduce consistently:

https://github.com/MetaMask/metamask-extension/assets/34557516/f73b9129-3ce9-4a74-9def-d4825c14e751

**Comment by adonesky1** [comment]
Sure enough. Reproducible. Here are the steps:

https://github.com/MetaMask/metamask-extension/assets/34557516/e53393f2-fcb3-49df-8bc0-14eec3aa54e5


Now I'll check if this happens on v12.0.0 before this patch

**Comment by adonesky1** [comment]
Yep its on v12.0.0

**Comment by adonesky1** [comment]
This patch is not causing the [bug shown in the comment above](https://github.com/MetaMask/metamask-extension/pull/25311#issuecomment-2168436635)

**Review by adonesky1 — APPROVED** [review]
LGTM!

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Comment by jiexi** [comment]
This very likely needs to be updated with the permissions-connect fix in this PR

https://github.com/MetaMask/metamask-extension/pull/25310/files

**Comment by jiexi** [comment]
Replaced by https://github.com/MetaMask/metamask-extension/pull/25451

**Pull request closed without merging**